### DISCUSSION TITLE
why the `publishNotReadyAddresses: true` can fix the split-brain for EMQX clusters

### DISCUSSION THREAD

**Discussion by wooonderful** [emqx/emqx-operator#414] [Q&A] [answered]
when I read the release notes of emqx-operator1.2.5, I find it fix the cluster's split-brain by setting `publishNotReadyAddresses: true`.

why this can fix the cluster split-brain?
What did emqx-operator or EMQX  do when set `publishNotReadyAddresses: true`.

<img width="709" alt="image" src="https://user-images.githubusercontent.com/115449150/195801881-89499665-d8b2-4a80-9689-93b1e2ecb043.png">

**Comment by Rory-Z** [comment, marked as answer]
EMQX's cluster discovery is done by querying the SRV records of the headless service to discover nodes. When `publishNotReadyAddresses: false`( this is default service config), the SRV records will be added to the headless service only when the Pod is ready, in this process, the records queried by EMQX's may not be consistent, for example, the first ready Pod only finds one SRV record, this is it's own SRV record.

When `publishNotReadyAddresses: true`, headless service will add the records of all the selector's Pods, regardless of whether the Pod is ready or not.

**Reply by wooonderful** [reply]
I see, thanks
